vscode-lean4/src/leanclient.ts

```diff
@@ -74,7 +74,7 @@ export class LeanClient implements Disposable {
 
     async restart(): Promise<void> {
         if (this.isStarted()) {
-            await this.stop()
+            void this.stop()
```

Comment:
This looks like a really dangerous idea (if stop was actually async).  I think you should just remove the async modifier on stop instead.